src/dawn/Optimizer/PassFieldVersioning.cpp

```diff
@@ -89,71 +91,165 @@ static void reportRaceCondition(const Statement& statement,
   }
 }
 
-} // anonymous namespace
+static void addAssignmentToDoMethod(std::unique_ptr<iir::DoMethod>& domethod, int assignmentID,
+                                    int assigneeID,
+                                    const std::shared_ptr<iir::StencilInstantiation>& context,
+                                    boost::optional<iir::Extents>& readExtents) {
+  // Create the StatementAccessPair of the assignment with the new and old variables
+  auto fa_assignee =
+      std::make_shared<FieldAccessExpr>(context->getMetaData().getNameFromAccessID(assigneeID));
+  auto fa_assignment =
+      std::make_shared<FieldAccessExpr>(context->getMetaData().getNameFromAccessID(assignmentID));
+  auto assignmentExpression = std::make_shared<AssignmentExpr>(fa_assignment, fa_assignee, "=");
+  auto expAssignment = std::make_shared<ExprStmt>(assignmentExpression);
+  auto assignmentStatement = std::make_shared<Statement>(expAssignment, nullptr);
+  auto pair = make_unique<iir::StatementAccessesPair>(assignmentStatement);
+  auto newAccess = std::make_shared<iir::Accesses>();
+
+  // The exetens were computed on a multistage-level and are passed to here. If they are
+  // uninitialized, we just copy the compute-domain, otherwise the read-extents of the versioned
+  // field determines how much copying needs to happen
+  if(readExtents.is_initialized()) {
+    newAccess->addWriteExtent(assignmentID, readExtents.get());
+    newAccess->addReadExtent(assigneeID, readExtents.get());
+  } else {
+    newAccess->addWriteExtent(assignmentID, iir::Extents(Array3i{{0, 0, 0}}));
+    newAccess->addReadExtent(assigneeID, iir::Extents(Array3i{{0, 0, 0}}));
+  }
+  pair->setAccesses(newAccess);
+  domethod->insertChild(std::move(pair));
+
+  // Add the new expressions to the map
+  context->getMetaData().insertExprToAccessID(fa_assignment, assignmentID);
+  context->getMetaData().insertExprToAccessID(fa_assignee, assigneeID);
+}
+/// @brief Creates the stage in which assignment happens (fill and flush)
```

Comment:
fill and flush are concepts of kcaches, so it is confusing to me to use it here in the doc

Reply:
changed the wording